Mathlib/Data/Matrix/Basis.lean

```diff
@@ -43,45 +43,52 @@ def stdBasisMatrix (i : m) (j : n) (a : α) : Matrix m n α := fun i' j' =>
 @[simp]
 theorem smul_stdBasisMatrix (i : m) (j : n) (a b : α) :
     b • stdBasisMatrix i j a = stdBasisMatrix i j (b • a) := by
-  unfold std_basis_matrix
+  unfold stdBasisMatrix
   ext
   simp
 #align matrix.smul_std_basis_matrix Matrix.smul_stdBasisMatrix
 
 @[simp]
 theorem stdBasisMatrix_zero (i : m) (j : n) : stdBasisMatrix i j (0 : α) = 0 := by
-  unfold std_basis_matrix
+  unfold stdBasisMatrix
   ext
   simp
 #align matrix.std_basis_matrix_zero Matrix.stdBasisMatrix_zero
 
 theorem stdBasisMatrix_add (i : m) (j : n) (a b : α) :
     stdBasisMatrix i j (a + b) = stdBasisMatrix i j a + stdBasisMatrix i j b := by
-  unfold std_basis_matrix; ext
+  unfold stdBasisMatrix; ext
   split_ifs with h <;> simp [h]
 #align matrix.std_basis_matrix_add Matrix.stdBasisMatrix_add
 
 theorem matrix_eq_sum_std_basis [Fintype m] [Fintype n] (x : Matrix m n α) :
-    x = ∑ (i : m) (j : n), stdBasisMatrix i j (x i j) := by
-  ext; symm
+    x = ∑ i : m, ∑ j : n, stdBasisMatrix i j (x i j) := by
+  ext i j; symm
   iterate 2 rw [Finset.sum_apply]
-  convert Fintype.sum_eq_single i _
-  · simp [std_basis_matrix]
-  · intro j hj
-    simp [std_basis_matrix, hj]
+  -- Porting note: was `convert`
+  rw [Fintype.sum_eq_single i
+    (?_ : (∀ j', j' ≠ i → ((∑ j : n, (stdBasisMatrix j' j (x j' j))) i j) = 0))]
+  · simp only [stdBasisMatrix]
+    rw [Fintype.sum_apply, Fintype.sum_apply]
+    simp
+  · intro j' hj'
+    simp only [stdBasisMatrix]
+    rw [Fintype.sum_apply, Fintype.sum_apply]
+    simp [hj']
 #align matrix.matrix_eq_sum_std_basis Matrix.matrix_eq_sum_std_basis
 
 -- TODO: tie this up with the `basis` machinery of linear algebra
 -- this is not completely trivial because we are indexing by two types, instead of one
 -- TODO: add `std_basis_vec`
 theorem std_basis_eq_basis_mul_basis (i : m) (j : n) :
     stdBasisMatrix i j 1 = vecMulVec (fun i' => ite (i = i') 1 0) fun j' => ite (j = j') 1 0 := by
-  ext
-  norm_num [std_basis_matrix, vec_mul_vec]
+  ext i' j'
+  simp only [stdBasisMatrix, vecMulVec, mul_ite, mul_one, mul_zero, of_apply]
+  simp_rw [@and_comm (i = i')]
```

Comment:
I'm not sure passing extra lemmas to `norm_num` actually does something. There's no good reason to use it here anyway, since it doesn't deal with numerals.